src/core/qgsabstractgeopdfexporter.cpp

```diff
@@ -123,7 +123,10 @@ bool QgsAbstractGeoPdfExporter::finalize( const QList<ComponentLayerDetail> &com
 
 QString QgsAbstractGeoPdfExporter::generateTemporaryFilepath( const QString &filename ) const
 {
-  return mTemporaryDir.filePath( filename );
+  QString cleanedFileName = filename;
+  cleanedFileName.replace( '\\', '_' );
+  cleanedFileName.replace( '/', '_' );
```

Comment:
wouldnt ':' need to be sanitized too (on Windows) ?

Reply:
Good point, I'll add

Reply:
https://github.com/qgis/QGIS/pull/51701

Reply:
(I'll merge this one and then backport the follow up)